### PR TITLE
SDIT-1161 update award now updates changed awards

### PR DESCRIPTION
**Pull request by andymarke**
TODO

- delete awards that are no longer referenced by DOS

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationService.kt [hunk 1]
```diff
@@ -727,13 +727,24 @@ class AdjudicationService(
     adjudicationNumber: Long,
     chargeSequence: Int,
     requests: CreateHearingResultAwardRequest,
-  ): CreateHearingResultAwardResponses {
-    // DPS does not associate award with result (migrated or synchronised data)
+  ): CreateHearingResultAwardResponses =
+    CreateHearingResultAwardResponses(
+      createHearingResultAwards(
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationService.kt:732]: extracted to a function and then used by update service

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/CreateHearingResultAwardResponse.kt [hunk 2]
```diff
@@ -8,12 +8,5 @@ import io.swagger.v3.oas.annotations.media.Schema
 data class CreateHearingResultAwardResponses(
 
   @Schema(description = "an ordered list of award response, the order matching the request order")
-  val awardResponses: List<CreateHearingResultAwardResponse>,
-)
-
-@JsonInclude(JsonInclude.Include.NON_NULL)
-@Schema(description = "Provides the generated Hearing Result Award composite ID after creation")
-data class CreateHearingResultAwardResponse(
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/CreateHearingResultAwardResponse.kt:16]: replaced this with the update version that is identical

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationsHearingResultAwardResourceIntTest.kt [hunk 3]
```diff
@@ -654,15 +676,15 @@ class AdjudicationsHearingResultAwardResourceIntTest : IntegrationTestBase() {
           .exchange()
           .expectStatus().isOk
           .expectBody()
-          .jsonPath("awardResponsesCreated[0].sanctionSequence").isEqualTo(4)
+          .jsonPath("awardResponsesCreated[0].sanctionSequence").isEqualTo(8)
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationsHearingResultAwardResourceIntTest.kt:679]: shows sanction sequence allows gaps (4 is not used but will choose 8 which is 1 plus highest)

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationService.kt [hunk 4]
```diff
@@ -785,77 +796,71 @@ class AdjudicationService(
...
-      AdjudicationHearingResultAward(
-        id = AdjudicationHearingResultAwardId(
-          offenderBookId = offenderBookId,
-          sanctionSequence = sanctionSeq + index,
+  private fun updateHearingResultAwards(
+    offenderBookId: Long,
+    adjudicationNumber: Long,
+    awardRequestsToUpdate: List<ExistingHearingResultAwardRequest>,
+  ) {
+    awardRequestsToUpdate.forEach {
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationService.kt:827]: main meat to PR, read a sanction and update it!

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**